io/plugins/eu.esdihumboldt.hale.io.codelist.skos.test/src/eu/esdihumboldt/hale/io/codelist/skos/test/SkosCodeListTest.java

```diff
@@ -0,0 +1,157 @@
+import eu.esdihumboldt.hale.common.codelist.CodeList.CodeEntry;
+import eu.esdihumboldt.hale.common.codelist.io.CodeListReader;
+import eu.esdihumboldt.hale.common.core.io.impl.LogProgressIndicator;
+import eu.esdihumboldt.hale.common.core.io.report.IOReport;
+import eu.esdihumboldt.hale.common.core.io.supplier.DefaultInputSupplier;
+import eu.esdihumboldt.hale.io.codelist.skos.reader.SkosCodeListReader;
+
+/**
+ * 
+ * SKOS Code list reader test
+ * 
+ * @author Arun
+ *
+ */
+public class SkosCodeListTest {
+
+	/**
+	 * test rdf file consisting only concepts
+	 * 
+	 * @throws Exception throws exception if something wrong occurs
+	 */
+	@Test
+	public void testSKOSFromRDF1() throws Exception {
+
+		CodeList codeList = readCodeList(getResourceURI("/data/test1.rdf"));
+
+		Collection<CodeEntry> entries = codeList.getEntries();
+		assertFalse(entries.isEmpty());
+
+		assertEquals(entries.size(), 1);
+
+		assertNotNull(codeList.getLocation());
+		assertNotNull(codeList.getIdentifier());
+
+		for (CodeEntry entry : entries) {
+			assertEquals("Data scientist", entry.getName());
+		}
+	}
+
+	/**
+	 * test rdf file consisting concept scheme and concepts
+	 * 
+	 * @throws Exception throws exception if something wrong occurs
+	 */
+	@Test
+	public void testSKOSFromRDF2() throws Exception {
+
+		CodeList codeList = readCodeList(getResourceURI("/data/test2.rdf"));
+
+		Collection<CodeEntry> entries = codeList.getEntries();
+		assertFalse(entries.isEmpty());
+
+		assertEquals(entries.size(), 3);
+
+		assertNotNull(codeList.getLocation());
+		assertNotNull(codeList.getIdentifier());
+		assertEquals("Types of aquifers.", codeList.getDescription());
+
+		List<String> concepts = Arrays.asList("confined subartesian", "confined artesian",
+				"unconfined");
+
+		for (CodeEntry entry : entries) {
+			assertTrue(concepts.contains(entry.getName()));
+		}
+
+	}
+
+	/**
+	 * 
+	 * test url 1 consisting concepts.
+	 * 
+	 * @throws Exception throws exception if something wrong occurs
+	 */
+	@Test
+	public void testSKOSFromURL1() throws Exception {
+
+		CodeList codeList = readCodeList(
+				URI.create("http://vocab.ndg.nerc.ac.uk/term/W080/1/CONT0001"));
+
+		Collection<CodeEntry> entries = codeList.getEntries();
+		assertFalse(entries.isEmpty());
+
+		assertEquals(entries.size(), 1);
+
+		assertNotNull(codeList.getLocation());
+		assertNotNull(codeList.getIdentifier());
+
+		for (CodeEntry entry : entries) {
+			assertEquals("Manufacturer", entry.getName());
+			assertEquals(
+					"Relevant contact information about the person or organisation that builds or develops a device or system.",
+					entry.getDescription());
+		}
+	}
+
+	/**
+	 * 
+	 * test url 2 consisting concepts (case: fallback).
+	 * 
+	 * @throws Exception throws exception if something wrong occurs
+	 */
+	@Test
+	public void testSKOSFromURL2() throws Exception {
+
+		CodeList codeList = readCodeList(
+				URI.create("http://vocab.ices.dk/services/rdf/collection/PARAM"));
```

Comment:
I don't think really relying on the online resource for that test makes sense. The online resource may change and thus the number of concepts or their definition.